cmd/describe/cluster/cmd.go

```diff
@@ -676,12 +684,14 @@ func formatCluster(cluster *cmv1.Cluster, scheduledUpgrade *cmv1.UpgradePolicy,
 		upgrade["nextRun"] = scheduledUpgrade.NextRun().Format("2006-01-02 15:04 MST")
 		ret["scheduledUpgrade"] = upgrade
 	}
+	ret["displayName"] = displayName
```

Comment:
Do we need to check if `displayName != ""`?

Reply:
I considered that but would prefer to show it as blank if it is

Reply:
This is the UI logic. I think we should do the same. Thoughts?
```
const getClusterName = (cluster: ClusterFromSubscription): string => {
  if (!cluster) {
    return '';
  }

  let clusterName = cluster.subscription?.display_name;
  if (!clusterName || clusterName === cluster.subscription?.external_cluster_id) {
    clusterName = cluster.name || cluster.external_id;
  }

  if (clusterName === undefined) {
    if (cluster.subscription?.status === 'Deprovisioned') {
      const subscriptionId = cluster.subscription?.id;
      if (subscriptionId) {
        return subscriptionId;
      }
    }
    return 'Unnamed Cluster';
  }
  return clusterName;
};
```

Reply:
As I mentioned on internal Slack, a lot of this is historic baggage that may be irrelevant; we just never had incentives to simplify it.  And it's so complex I'd hate to enshrine it as the official way one should consume the API :-1:.

For example the `display_name === subscription?.external_cluster_id` case falling back to `cluster.name || cluster.external_id` — I've been on the cluster_mgmt backend + UI team from the very start yet I can't explain in what scenario that is supposed to help... :laughing:  
Normal OCP clusters have cluster.name === cluster.display_name === cluster.external_id === subscription.display_name === subscription.external_cluster_id, so you'll get that same UUID anyway (until user edits display_name).

But here the decision is simpler: You're going to show the fields separately:
```
		"Name:                       %s\n"+
		"Display Name:               %s\n"+
                ...
		"External ID:                %s\n"+
```
so if display_name is absent/blank, IMHO the appropriate thing is just reflect API reality and show it empty.  Let user figure out if other fields contain relevant info but don't lie to them that it has a display_name when it doesn't.

----

I do see some OCP & OCP-AssistedInstall subscriptions in prod with `display_name = '' OR display_name IS NULL`, and I'm gonna reply here with more details soon.  
But none for ROSA — all MOA & MOA-HostedControlPlane clusters have a non-empty display_name :heavy_check_mark:

Reply:
I tend to agree with showing a blank display name if not available, to keep the logic in the CLI simple.

Reply:
[just following up as promised, but nothing here is relevant to `rosa`]

So I didn't find a single clear pattern to OCP & OCP-AssistedInstall subscriptions with null blank (67 cases) display_name; many Archived & old but not all, various provenance, various states... some recent, including 1 Active now.  
8 even had non-UUID _cluster.display_name_!  But those were all old <=2021.

What I did find is API allows users to set a blank display_name:
```
echo '{"display_name": ""}' | ocm patch sub 2aiKK76moYZywHSpfwV4m0BzvEZ
```
upon which the field disappears from the API (omitempty), until you set it non-blank again.  This could be what happened to the 1 Active?

It can't explain the IS NULL ones.

* Well 334 of the 335 IS NULL ones are all OCP that also lack subscription.status (match `status = ''`), which is extra weird :question:.  
  The one exception (OCP-AssistedInstall, Reserved) seems to have been some manual test (has "XXX" at end of cluster_id and external_cluster_id) so can be ignored.